<!-- 
    ROS navigation stack with velocity smoother and safety (reactive) controller
    http://wiki.ros.org/move_base
-->
<launch>
  <include file="$(find turtlebot_navigation)/launch/includes/velocity_smoother.launch.xml"/>
  <include file="$(find turtlebot_navigation)/launch/includes/safety_controller.launch.xml"/>
  
  <arg name="odom_topic" default="odom" />
  <arg name="laser_topic" default="scan" />

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find upc_mrn)/param/move_base_params.yaml" command="load" />

    <rosparam file="$(find upc_mrn)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find upc_mrn)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />

    <rosparam file="$(find upc_mrn)/param/local_costmap_params.yaml" command="load" />   
    <rosparam file="$(find upc_mrn)/param/global_costmap_params.yaml" command="load" />

    <rosparam file="$(find upc_mrn)/param/dwa_local_planner_params.yaml" command="load" />
    <rosparam file="$(find upc_mrn)/param/base_local_planner_params.yaml" command="load" />

    <rosparam file="$(find upc_mrn)/param/global_planner_params.yaml" command="load" />
    <rosparam file="$(find upc_mrn)/param/navfn_global_planner_params.yaml" command="load" />

    <remap from="cmd_vel" to="navigation_velocity_smoother/raw_cmd_vel"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg laser_topic)"/>
  </node>
</launch>